zgw/taak/src/main/kotlin/nl/nlportal/zgw/taak/service/TaakService.kt

```diff
@@ -105,23 +148,93 @@ open class TaakService(
+
+        val updateRequest = UpdateObjectsApiObjectRequest.fromObjectsApiObject(objectsApiTask)
+        updateRequest.record.data.formtaak?.verzondenData = submissionAsMap
+        updateRequest.record.data.status = TaakStatus.AFGEROND
+        updateRequest.record.correctedBy = authentication.getUserRepresentation()
+        updateRequest.record.correctionFor = objectsApiTask.record.index.toString()
+
+        val updatedObjectsApiTask = objectsApiClient.updateObject(objectsApiTask.uuid, updateRequest)
+        return TaakV2.fromObjectsApi(updatedObjectsApiTask)
+    }
+
+    private suspend inline fun <reified T> getObjectsApiTaak(
         taskId: UUID,
-        authentication: PortalAuthentication,
-    ): ObjectsApiObject<TaakObject> {
+        authentication: CommonGroundAuthentication,
+        objectTypeUrl: String,
+    ): ObjectsApiObject<T> {
         val userSearchParameters = getUserSearchParameters(authentication)
         val taskIdSearchParameter = ObjectSearchParameter("verwerker_taak_id", Comparator.EQUAL_TO, taskId.toString())
 
-        return objectsApiClient.getObjects<TaakObject>(
+        return objectsApiClient.getObjects<T>(
             objectSearchParameters = userSearchParameters + taskIdSearchParameter,
-            objectTypeUrl = objectsApiTaskConfig.typeUrl,
+            objectTypeUrl = objectTypeUrl,
             page = 1,
             pageSize = 2,
         ).results.single()
     }
 
-    private fun getUserSearchParameters(authentication: PortalAuthentication): List<ObjectSearchParameter> {
-        return createIdentificatieSearchParameters(authentication.userType, authentication.userId)
+    private suspend inline fun <reified T> getTakenResultPage(
+        pageNumber: Int,
+        pageSize: Int,
+        authentication: CommonGroundAuthentication,
+        zaakUUID: UUID? = null,
+        objectTypeUrl: String,
+    ): ResultPage<ObjectsApiObject<T>> {
+        val objectSearchParameters = mutableListOf<ObjectSearchParameter>()
+
+        objectSearchParameters.addAll(getUserSearchParameters(authentication))
+        objectSearchParameters.add(ObjectSearchParameter("status", Comparator.EQUAL_TO, "open"))
+
+        zaakUUID?.let {
+            objectSearchParameters.add(
+                ObjectSearchParameter(
+                    "zaak",
+                    Comparator.STRING_CONTAINS,
+                    it.toString(),
+                ),
+            )
+        }
+        return objectsApiClient.getObjects<T>(
+            objectSearchParameters = objectSearchParameters,
+            objectTypeUrl = objectTypeUrl,
+            page = pageNumber,
+            pageSize = pageSize,
+            ordering = "-record__startAt",
+        )
+    }
+
+    private fun getUserSearchParameters(authentication: CommonGroundAuthentication): List<ObjectSearchParameter> {
+        return when (authentication) {
```

Comment:
Je kan hier ook return `createIdentificatieSearchParameters(authentication.userType, authentication.userId)` doen. Dan doet KeycloakUserAuthentication ook mee namelijk

Als development al in deze branch zit.